core/lib/contracts/src/lib.rs

```diff
@@ -288,12 +292,50 @@ pub fn read_zbin_bytecode(relative_zbin_path: impl AsRef<Path>) -> Vec<u8> {
     read_zbin_bytecode_from_path(bytecode_path)
 }
 
+pub fn read_yul_bytecode(relative_artifacts_path: PathBuf, name: &str) -> Vec<u8> {
+    let artifacts_path = Path::new(&home_path()).join(relative_artifacts_path);
+    read_yul_bytecode_by_path(artifacts_path, name)
+}
+
+pub fn read_yul_bytecode_by_path(artifacts_path: PathBuf, name: &str) -> Vec<u8> {
+    let bytecode_path = artifacts_path.join(format!("{0}.yul/{0}.yul.zbin", name));
+
+    if fs::exists(&bytecode_path).unwrap_or_else(|_| panic!("Invalid path: {:?}", &bytecode_path)) {
+        read_zbin_bytecode_from_path_utf8(bytecode_path)
+    } else {
+        let bytecode_path_legacy = artifacts_path.join(format!("{}.yul.zbin", name));
+
+        if fs::exists(&bytecode_path_legacy)
+            .unwrap_or_else(|_| panic!("Invalid path: {:?}", &bytecode_path_legacy))
+        {
+            read_zbin_bytecode_from_path(bytecode_path_legacy)
+        } else {
+            panic!(
+                "Can't find bytecode for '{}' yul contract at {:?}",
+                name, artifacts_path
+            )
+        }
+    }
+}
+
 /// Reads zbin bytecode from a given path.
 fn read_zbin_bytecode_from_path(bytecode_path: PathBuf) -> Vec<u8> {
     fs::read(&bytecode_path)
         .unwrap_or_else(|err| panic!("Can't read .zbin bytecode at {:?}: {}", bytecode_path, err))
 }
 
+/// Reads zbin bytecode from a given path as utf8 text file.
+fn read_zbin_bytecode_from_path_utf8(bytecode_path: PathBuf) -> Vec<u8> {
+    let buffer = fs::read(&bytecode_path)
+        .unwrap_or_else(|err| panic!("Can't read .zbin bytecode at {:?}: {}", bytecode_path, err));
+
+    hex_string_to_bytes(
+        &String::from_utf8(buffer)
```

Comment:
This performs an unnecessary allocation (`str::from_utf8` is a non-allocating counterpart). But it's not necessary if you use `hex` crate in any case, since it allows parsing directly from `&[u8]`.